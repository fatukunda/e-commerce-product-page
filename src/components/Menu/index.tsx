interface IMenuProps {
  classList?: string;
  children?: JSX.Element | JSX.Element[];
}

const Menu = ({ classList, children }: IMenuProps): JSX.Element => {
  return <div className={`absolute top-[70px] overflow-auto ${classList}`}>{children}</div>;
};

export default Menu;
