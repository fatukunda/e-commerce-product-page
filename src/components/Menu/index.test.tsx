import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Provider as ReduxProvider } from "react-redux";

import Menu from ".";
import mockStore from "../mocks/store";

const renderMenu = () =>
  render(
    <ReduxProvider store={mockStore()}>
      <Menu />
    </ReduxProvider>
  );
describe("Tests for <Menu/>", () => {
  afterEach(cleanup);
  test("Menu: Snapshot", () => {
    const component = renderer.create(
      <ReduxProvider store={mockStore()}>
        <Menu />
      </ReduxProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Renders without crashing", () => {
    renderMenu();
  });
});
