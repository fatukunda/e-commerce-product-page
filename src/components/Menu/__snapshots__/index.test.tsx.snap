// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Menu/> Menu: Snapshot 1`] = `
<div
  className="absolute top-[70px] overflow-auto undefined"
/>
`;
